File: frontend/lib/models/zone_vol.dart
class ZoneDeVol {
  final int? zoneId;
  final String nomZone;
  final String typeZone;
  final int risque;
  final String? restrictions;

  ZoneDeVol({
    this.zoneId,
    required this.nomZone,
    required this.typeZone,
    required this.risque,
    this.restrictions,
  });

  Map<String, dynamic> toJson() {
    return {
      'nom_zone': nomZone,
      'type_zone': typeZone,
      'risque': risque,
      if (restrictions != null) 'restrictions': restrictions,
    };
  }

  factory ZoneDeVol.fromJson(Map<String, dynamic> json) {
    return ZoneDeVol(
      zoneId: json['zone_id'],
      nomZone: json['nom_zone'],
      typeZone: json['type_zone'],
      risque: json['risque'],
      restrictions: json['restrictions'],
    );
  }
}

enum TypeZone {
  urbaine('urbaine', 'Urbaine'),
  rurale('rurale', 'Rurale'),
  industrielle('industrielle', 'Industrielle'),
  residentielle('residentielle', 'Résidentielle'),
  commerciale('commerciale', 'Commerciale'),
  militaire('militaire', 'Militaire'),
  hopital('hopital', 'Hôpital');

  final String value;
  final String label;
  const TypeZone(this.value, this.label);
}

File: backend/app/models/zone_vol.py
from sqlalchemy import Column, Integer, String, Text
from sqlalchemy.orm import relationship
from app.database import Base
import enum


class TypeZone(str, enum.Enum):
    URBAINE = "urbaine"
    RURALE = "rurale"
    INDUSTRIELLE = "industrielle"
    RESIDENTIELLE = "residentielle"
    COMMERCIALE = "commerciale"
    MILITAIRE = "militaire"
    HOPITAL = "hopital"


class ZoneDeVol(Base):
    __tablename__ = "zones_de_vol"

    zone_id = Column(Integer, primary_key=True, index=True)
    nom_zone = Column(String(255), nullable=False)  # nom de la zone (ex: Zone A, Zone B)
    type_zone = Column(String(255), nullable=False)  # type de zone
    risque = Column(Integer, nullable=False)  # niveau de risque dans cette zone (échelle de 1 à 5)
    restrictions = Column(Text, nullable=True)  # détails sur les restrictions spécifiques à la zone

    # Relations
    missions = relationship("Mission", back_populates="zone")